### PR TITLE
Functions - Prime Checker Function

### PR DESCRIPTION
**Pull request by Abduliitr**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [x] Author
    - [x] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [x] Author
    - [x] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [x] Author
    - [x] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [x] Author
    - [x] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [x] There are at least 2 test cases
    - [x] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

File: solutions/functions/prime-checker-function.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+
+// Following function takes integer and should return True if it's prime 
+// otherwise  should return False.
+function is_prime(input_number){
```
Comment (shivateja1729) [inline solutions/functions/prime-checker-function.js:15]: In the template, the functions  were in camelCase, but this solution uses snake_case for representing functions and variables. As per the coding conventions mentioned, we only use camelCase for functions and variables and PascalCase for classes.

Comment (LVSsandeepVarma) [reply solutions/functions/prime-checker-function.js:15]: I have updated this template code

Comment (Abduliitr) [reply solutions/functions/prime-checker-function.js]: Okay updated the code as per the latest template.

**Review by shivateja1729 — APPROVED** [review]

File: solutions/functions/prime-checker-function.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+// Following function takes integer and should return True if it's prime 
+// otherwise  should return False.
+function isPrime(inputNumber){
+    // You can start below this
+    
+    if(inputNumber <= 1) return "False";
+	else if(inputNumber === 2) return "True";
+
+    for(let i = 2; i <= parseInt(Math.sqrt(inputNumber)); i++){
```
Comment (sailesh-ramayanam) [inline solutions/functions/prime-checker-function.js:21]: 1. Please move `parseInt(Math.sqrt(inputNumber))` outside of `for` loop. In compiled languages (like `C, C++, Java`), there is a high chance that compiler will optimize this, but I am not sure of `Javascript`, which is an interpreted language.
2. Also, please us `Math.floor`

**Review by sailesh-ramayanam — CHANGES_REQUESTED** [review]

File: solutions/functions/prime-checker-function.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+// -------- Do NOT edit anything above this line ----------
+
+// Following function takes integer and should return True if it's prime 
+// otherwise  should return False.
+function isPrime(inputNumber){
+    // You can start below this
+    
+    if(inputNumber <= 1) return "False";
+	else if(inputNumber === 2) return "True";
```
[the remark this answers is not in the document]
Comment (Abduliitr) [reply solutions/functions/prime-checker-function.js:19]: Addressed.

File: solutions/functions/prime-checker-function.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+// -------- Do NOT edit anything above this line ----------
+
+// Following function takes integer and should return True if it's prime 
+// otherwise  should return False.
+function isPrime(inputNumber){
+    // You can start below this
+    
+    if(inputNumber <= 1) return "False";
```
[the remark this answers is not in the document]
Comment (Abduliitr) [reply solutions/functions/prime-checker-function.js:18]: Addressed.

File: solutions/functions/prime-checker-function.js (hunk 2 above)
Comment (Abduliitr) [reply solutions/functions/prime-checker-function.js:21]: Addressed

**Review by sailesh-ramayanam — APPROVED** [review]

**Comment by sailesh-ramayanam** [comment]
@shivateja1729 Please correct the example. It must be a concrete example. 
image

**Comment by shivateja1729** [comment]
@sailesh-ramayanam  Updated. Please have a look at it.
Screenshot 2022-03-08 223458
.

**Comment by sailesh-ramayanam** [comment]
@shivateja1729 `True` and `False` - first letter must be upper case.

**Comment by shivateja1729** [comment]
> @shivateja1729 `True` and `False` - first letter must be upper case.

@sailesh-ramayanam I have made changes.

**Pull request merged**